### PR TITLE
More detailed error logging on segment activities

### PR DESCRIPTION
**Pull request by drcrallen**

### CODE REVIEW

**Comment by nishantmonu51** [comment]
+1

**Comment by drcrallen** [comment]
Java 8 + zk atomicity seems keeps breaking, trying for a better travis CI run

File: extensions/s3-extensions/src/main/java/io/druid/storage/s3/S3DataSegmentKiller.java [hunk 1]
```diff
@@ -63,7 +63,7 @@ public void kill(DataSegment segment) throws SegmentLoadingException
       }
     }
     catch (ServiceException e) {
-      throw new SegmentLoadingException(e, "Couldn't kill segment[%s]", segment.getIdentifier());
+      throw new SegmentLoadingException(e, "Couldn't kill segment[%s]: [%s]", segment.getIdentifier(), e);
```
Comment (himanshug) [inline extensions/s3-extensions/src/main/java/io/druid/storage/s3/S3DataSegmentKiller.java]: "e" is already set as the exception "cause", why do we need to put e.toString() in the message as well or did you want to put e.getMessage() ?

Comment (drcrallen) [inline extensions/s3-extensions/src/main/java/io/druid/storage/s3/S3DataSegmentKiller.java]: E as cause only captures stack trace. E as parameters has a lot of info in its toString

**Pull request merged**